azkaban-common/src/test/java/azkaban/executor/DispatcherTest.java

```diff
@@ -0,0 +1,635 @@
+      result = mFilter.check(this.getExecutorByName("Executor3"), dispatchingObj);
+      // expect false, for priority.
+      /*
+      2 [main] INFO azkaban.executor.dispatcher.CandidateFilter  - start checking 'Executor3' with factor filter for 'Mockfilter'
+      2 [main] INFO azkaban.executor.dispatcher.CandidateFilter  - [Factor: requiredRemainingMemory] filter result : true
+      2 [main] INFO azkaban.executor.dispatcher.CandidateFilter  - [Factor: requiredProprity] filter result : false
+      2 [main] INFO azkaban.executor.dispatcher.CandidateFilter  - Final checking result : false
+      */
+      Assert.assertFalse(result);
+
+      // add the remaining space filter.
+      mFilter.registerFilterforRemainingTmpSpace();
+
+      // expect pass.
+      result = mFilter.check(this.getExecutorByName("Executor2"), dispatchingObj);
+      /*
+      3 [main] INFO azkaban.executor.dispatcher.CandidateFilter  - start checking 'Executor2' with factor filter for 'Mockfilter'
+      3 [main] INFO azkaban.executor.dispatcher.CandidateFilter  - [Factor: requiredRemainingMemory] filter result : true
+      3 [main] INFO azkaban.executor.dispatcher.CandidateFilter  - [Factor: requiredRemainingTmpSpace] filter result : true
+      3 [main] INFO azkaban.executor.dispatcher.CandidateFilter  - [Factor: requiredProprity] filter result : true
+      3 [main] INFO azkaban.executor.dispatcher.CandidateFilter  - Final checking result : true
+      */
+      Assert.assertTrue(result);
+
+      // expect false, remaining tmp, priority will also fail but the logic shortcuts when the Tmp size check Fails.
+      result = mFilter.check(this.getExecutorByName("Executor8"), dispatchingObj);
+      /*
+      4 [main] INFO azkaban.executor.dispatcher.CandidateFilter  - start checking 'Executor8' with factor filter for 'Mockfilter'
+      4 [main] INFO azkaban.executor.dispatcher.CandidateFilter  - [Factor: requiredRemainingMemory] filter result : true
+      4 [main] INFO azkaban.executor.dispatcher.CandidateFilter  - [Factor: requiredRemainingTmpSpace] filter result : false
+      4 [main] INFO azkaban.executor.dispatcher.CandidateFilter  - Final checking result : false
+      */
+      Assert.assertFalse(result);
+
+  }
+
+  @Test
+  public void testExecutorComparer() throws Exception {
+    MockComparator comparator = new MockComparator();
+    comparator.registerComparerForMemory(5);
+
+    MockExecutorObject nextExecutor = Collections.max(this.executorList, comparator);
+
+    // expect the first item to be selected, memory wise it is the max.
+    Assert.assertEquals(this.getExecutorByName("Executor11"),nextExecutor);
+
+    // add the priority factor.
+    // expect again the #9 item to be selected.
+    comparator.registerComparerForPriority(6);
+    nextExecutor = Collections.max(this.executorList, comparator);
+    /*
+      2 [main] INFO azkaban.executor.dispatcher.CandidateComparator  - start comparing 'Executor8' with 'Executor2',  total weight = 5 
```

Comment:
I wonder if it makes sense to include these comments in the test because if they change tomorrow, then they are out of dated.